### PR TITLE
Pro-forma PR for @sebasje's libsplit branch against master

### PR DESCRIPTION
**Pull request by danimo**

### CODE REVIEW

File: src/libclient/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,175 @@
+project(libowncloudclient)
```
Comment (danimo) [inline src/libclient/CMakeLists.txt]: lib must not use the name owncloud (due to theming), let's rather use libui or something more descriptive (analogous to `libsync`).

File: src/libclient/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+#
+if ( APPLE )
+    list(APPEND OS_SPECIFIC_LINK_LIBRARIES
+         /System/Library/Frameworks/CoreServices.framework
+         /System/Library/Frameworks/Foundation.framework
+         /System/Library/Frameworks/AppKit.framework
+    )
+endif()
+
+set(owncloudclientlib_NAME "owncloudclient")
```
Comment (danimo) [inline src/libclient/CMakeLists.txt]: see above.

**Comment by danimo** [comment]
What is your intention with breaking up those libraries? Being able to have a private fork (i.e. by using the theming option) is fine, but at least I would still be reluctant to commit to a stable API.

File: src/libclient/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+#     if(NEON_WITH_LFS)
+#         add_definitions(-DNE_LFS)
+#     endif()
+# endif()
+#
+# if(ZLIB_FOUND)
+#     list(APPEND libowncloudclient_LINK_TARGETS ${ZLIB_LIBRARIES})
+#     include_directories(${ZLIB_INCLUDE_DIRS})
+# endif(ZLIB_FOUND)
+#
```
Comment (dragotin) [inline src/libclient/CMakeLists.txt]: Why are these checks commented?

**Comment by dragotin** [comment]
I think the directory name "libclient" is not good, not specific enough. Maybe something like libsyncenvironment (yes, I know, too long...) ?

Also, we must not forget to verifiy the RPATH setting since we install that library to a application directory in `/usr/lib*`, right? 

And finally: We include `csync_private.h` too much. It was intended to be... well private :-) We should reconsider to rename it and/or reinvestigate which parts of it really need to be public.

File: src/libclient/CMakeLists.txt (hunk 1 above)
Comment (sebasje) [inline src/libclient/CMakeLists.txt]: libsyncui? libui is too generic, IMO.

File: src/libclient/CMakeLists.txt (hunk 3 above)
Comment (sebasje) [inline src/libclient/CMakeLists.txt]: They're not needed in this library, I might as well remove them there.

**Comment by sebasje** [comment]
From an earlier email exchange with Klaas and Olivier:

My goal is to use as much as possible of the existing 
owncloud client's code and put a Plasma UI (and perhaps some deeper 
integration bits) on top of it.

My approach is to create a headless synching client that runs in the 
background (autostarted in Plasma) and have a Plasmoid and a Systemsettings 
module communicate with it. This allows us to put different UIs for different 
devices and make owncloud synching behave more like a service provided by the 
system rather than an application in the traditional sense.

File: src/libclient/CMakeLists.txt (hunk 3 above)
Comment (ogoffart) [inline src/libclient/CMakeLists.txt]: Yes, they can be removed.

**Comment by ogoffart** [comment]
the include csync_private can hopefully disapear in the future.  Ideally, the SyncEngine would create the csync_context on demand.  
The optimisation to keep the csync_context between sync is not worth it anymore since we do not reload the owncloud plugin all the time and such.
But for now i think it's fine to include it

File: src/libclient/CMakeLists.txt (hunk 1 above)
Comment (ogoffart) [inline src/libclient/CMakeLists.txt]: what's wront with the `libowncloudclient` name?  It is an internal name only so branding should not care.  Or am i missing something.

Comment (dragotin) [inline src/libclient/CMakeLists.txt]: I agree with @sebasje, `libsyncui` is cool. We really want to avoid the word "owncloud" here.

Comment (ogoffart) [inline src/libclient/CMakeLists.txt]: how about  `libsyncclient`?
(maybe we can rename the other lib in `libsyncengine`

**Comment by ogoffart** [comment]
@sebasje  will you have time to address the issue that @danimo raised?

I think we should merge this sooner than later if we want it.

**Comment by sebasje** [comment]
Yes, I'll try to make some time shortly.

It would be good if someone could test my changes on WIndows and Mac OS, I don't have tese OSes available.

**Comment by ogoffart** [comment]
ping?

**Comment by sebasje** [comment]
I haven't forgotten about it, it just slipped down my list a bit with some other projects going on. I'll try to update my branch shortly and will let you have another go at it.

Thanks for the nudge. :)

**Comment by danimo** [comment]
This is obsolete now

**Pull request closed without merging**